buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/RestResourcesPlugin.java

```diff
@@ -86,21 +86,29 @@
     public void apply(Project project) {
         RestResourcesExtension extension = project.getExtensions().create(EXTENSION_NAME, RestResourcesExtension.class);
 
+        // tests
         Provider<CopyRestTestsTask> copyRestYamlTestTask = project.getTasks()
             .register("copyYamlTestsTask", CopyRestTestsTask.class, task -> {
                 task.includeCore.set(extension.restTests.getIncludeCore());
                 task.includeXpack.set(extension.restTests.getIncludeXpack());
-                task.coreConfig = project.getConfigurations().create("restTest");
+                task.coreConfig = project.getConfigurations().create("restTestConfig");
+                project.getConfigurations().maybeCreate("restTests");
+                project.getConfigurations().maybeCreate("restXpackTests");
```

Comment:
I think we should just use `create` instead of `maybeCreate` and remove any instances in projects that are manually creating these. The risk with the lenient approach is that some project declares this configuration for one purpose, and we see it and use it for a different one. This should be explicit. If you create a conflicting configuration we should error.

Reply:
good call, implemented on https://github.com/elastic/elasticsearch/pull/53620/commits/287fb72ec4ce5984e41cd69de907b1092863b200